python/triton/compiler.py

```diff
@@ -990,19 +990,20 @@ def _compile(fn, signature: str, device: int = -1, constants=dict(), specializat
     llvm_ir = make_llvm_ir(module)
 
     assert device >= 0, "device should be provided."
-    ptxas, cuda_version = path_to_ptxas()
-    compute_capability = torch.cuda.get_device_capability(device)
-    compute_capability = compute_capability[0] * 10 + compute_capability[1]
-    ptx_version = ptx_get_version(cuda_version)
-    ptx = make_ptx(llvm_ir, compute_capability, ptx_version)
-    shem_size = _triton.get_shared_memory_size(module)
-    kernel_name = ptx_get_kernel_name(ptx)
-    if output == "ptx":
-        return ptx, shem_size, kernel_name
-
-    cubin = make_cubin(ptx, ptxas, compute_capability)
-    if output == "cubin":
-        return cubin, ptx, shem_size, kernel_name
+    if torch.version.hip is None:
+        ptxas, cuda_version = path_to_ptxas()
+        compute_capability = torch.cuda.get_device_capability(device)
+        compute_capability = compute_capability[0] * 10 + compute_capability[1]
+        ptx_version = ptx_get_version(cuda_version)
+        ptx = make_ptx(llvm_ir, compute_capability, ptx_version)
+        shem_size = _triton.get_shared_memory_size(module)
+        kernel_name = ptx_get_kernel_name(ptx)
+        if output == "ptx":
+            return ptx, shem_size, kernel_name
+
+        cubin = make_cubin(ptx, ptxas, compute_capability)
+        if output == "cubin":
+            return cubin, ptx, shem_size, kernel_name
```

Comment:
@dfukalov From the look of it, the path from LLVM IR to AMDGCN ISA and AMD HSA code object is not done yet?

Please advise if there is any end-to-end example which runs now?

If not, there are quite a few pointers you may follow:
- Triton LLVM backend . @micmelesse should possess knowledge on how that's implemented.
- Clang HIP toolchain
- TensorFlow XLA
- rocMLIR
- ROCm TVM

Reply:
There is no end-to-end built and ran example yet. The PR is a just preparation part to support.